server/src/main/java/org/opensearch/cluster/metadata/MetadataCreateIndexService.java

```diff
@@ -1002,6 +1005,52 @@ private static void updateRemoteStoreSettings(Settings.Builder settingsBuilder,
         }
     }
 
+    /**
+     * Updates index settings to enable remote store by default for migration towards remote store backed clusters
+     * @param settingsBuilder index settings builder to be updated with relevant settings
+     * @param clusterState state of cluster
+     * @param clusterSettings cluster level settings
+     */
+    public static void updateRemoteStoreSettingsForMigration(
+        Settings.Builder settingsBuilder,
+        ClusterState clusterState,
+        ClusterSettings clusterSettings
+    ) {
+        String value = settingsBuilder.get(SETTING_REMOTE_STORE_ENABLED);
+        if (value != null && value.toLowerCase(Locale.ROOT).equals("true")) {
+            return;
+        }
+
+        boolean isMixedMode = clusterSettings.get(RemoteStoreNodeService.REMOTE_STORE_COMPATIBILITY_MODE_SETTING)
+            .equals(RemoteStoreNodeService.CompatibilityMode.MIXED);
+        boolean isRemoteStoreMigrationDirection = clusterSettings.get(RemoteStoreNodeService.MIGRATION_DIRECTION_SETTING)
+            .equals(RemoteStoreNodeService.Direction.REMOTE_STORE);
+
```

Comment:
can we create a method in `RemoteStoreNodeService` like `isMigratingToRemoteStore` and reuse in future?